tests/src/test/scala/cats/tests/RegressionSuite.scala

```diff
@@ -115,7 +115,7 @@ class RegressionSuite extends CatsSuite {
     // shouldn't have ever evaluted validate(8)
     checkAndResetCount(3)
 
-    LazyList(1, 2, 6, 8).traverse(validate) should ===(Either.left("6 is greater than 5"))
+    Stream(1, 2, 6, 8).traverse(validate) should ===(Either.left("6 is greater than 5"))
```

Comment:
I don't know much of these set of regression tests, but are we sure that `LazyList` doesn't need them (couldn't find the `LazyList` version)?

Reply:
I omitted these intentionally, but looking more closely I'm surprised the property being tested isn't covered by the `Traverse` laws, and we probably do want it for `LazyList`. Will add now.